src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1576,19 +1596,6 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
             _connection.TerminalOutput(_connectionOutputEventToken);
             _connectionStateChangedRevoker.revoke();
 
-            // Clear out the cursor timer, so it doesn't trigger again on us once we're destructed.
-            if (auto localCursorTimer{ std::exchange(_cursorTimer, std::nullopt) })
-            {
-                localCursorTimer->Stop();
-                // cursorTimer timer, now stopped, is destroyed.
-            }
-
-            if (auto localAutoScrollTimer{ std::exchange(_autoScrollTimer, nullptr) })
-            {
-                localAutoScrollTimer.Stop();
-                // _autoScrollTimer timer, now stopped, is destroyed.
-            }
-

```

Comment:
i'm surprised tearing down the timers isn't necessary anymore, but I'm willing to believe it :smile:

Reply:
Those `Stop` calls were only throwing in the event that `TermControl` was being destructed at the end of an event handler, which was strange. Could just wrap them in try-catch blocks if the `Stop` calls really should be there 🤔

Reply:
The most correct solution might be to make both the timers `Tick` events take a `get_weak()` as opposed to `this`, right? If we had done that from the start, we probably wouldn't have ever needed these lines...